### PR TITLE
Order nwp

### PR DESCRIPTION
**Pull request by peterdudfield**
# Pull Request

## Description

Make sure nwp batch.example object always has the same order of dimensions

## How Has This Been Tested?

normal unittests

- [ ] No
- [x] Yes

## Checklist:

- [x] My code follows [OCF's coding style guidelines](https://github.com/openclimatefix/nowcasting/blob/main/coding_style.md)
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

**Comment by peterdudfield** [comment]
This could also be done for satellite data- but I see @JackKelly is working on that at the moment, so perhaps it can be done at a later time

File: nowcasting_dataset/data_sources/nwp/nwp_model.py [hunk 1]
```diff
@@ -23,8 +23,9 @@ def model_validation(cls, v):
 
         v.check_nan_and_inf(data=v.data)
 
-        v.check_data_var_dim(
-            v.data, ("example", "time_index", "x_osgb_index", "y_osgb_index", "channels_index")
-        )
+        dims = ("example", "time_index", "x_osgb_index", "y_osgb_index", "channels_index")
+        v.check_data_var_dim(v.data, dims)
+
+        v.__setitem__("data", v.data.transpose(*dims))
```
Comment (peterdudfield) [inline nowcasting_dataset/data_sources/nwp/nwp_model.py:29]: could actually add this to 'check_data_var_dim' then it would be neater

**Review by JackKelly — COMMENTED** [review]
Sorry I'm a bit late! One tiny comment!

**Comment by peterdudfield** [comment]
im happy with that. Thats how pytorch seems to order them too

File: nowcasting_dataset/data_sources/nwp/nwp_model.py [hunk 2]
```diff
@@ -23,8 +23,8 @@ def model_validation(cls, v):
 
         v.check_nan_and_inf(data=v.data)
 
-        v.check_data_var_dim(
-            v.data, ("example", "time_index", "x_osgb_index", "y_osgb_index", "channels_index")
-        )
+        dims = ("example", "time_index", "x_osgb_index", "y_osgb_index", "channels_index")
```
[the remark this answers is not in the document]
Comment (peterdudfield) [reply nowcasting_dataset/data_sources/nwp/nwp_model.py:26]: > This looks good, although I'd advocate for swapping the order of `x_osgb_index` and `y_osgb_index`... in general, I think `xarray` (and some other libraries) actually expect the order of `x` and `y` to be `(y, x)` in order to plot correctly etc., so I wonder if our code should standardise on ordering the spatial coordinates `y, x`? Any thoughts?

moved into here now https://github.com/openclimatefix/nowcasting_dataset/pull/590

**Pull request merged**